src/providers/WorkflowCore.Persistence.MongoDB/Services/MongoPersistenceProvider.cs

```diff
@@ -315,9 +315,9 @@ public async Task ScheduleCommand(ScheduledCommand command)
             {
                 await ScheduledCommands.InsertOneAsync(command);
             }
-            catch (MongoBulkWriteException ex)
+            catch (MongoWriteException ex)
```

Comment:
Can we check for both types?

Reply:
Future proofing? Done.

The Mongo C# driver converts any `MongoBulkWriteException` thrown by `InsertOneAsync` into a `MongoWriteException` (https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver/MongoCollectionBase.cs#L501). Exceptions thrown are not documented as part of the method API, so that can change.

Reply:
Actually, I think it did change at some point... when I was debugging with an older version I get MongoBulkWriteException.